PiggyBank/Domain/UseCases/ChangeAndUpdateAccountsUseCase.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+
+final class ChangeAndUpdateAccountsUseCase {
+    
+    private let createUpdateAccountUseCase: CreateUpdateAccountUseCase
+    private let getAccountsUseCase: GetAccountsUseCase
+    
+    private let semaphore = DispatchSemaphore(value: 0)
+
+    init(createUpdateAccountUseCase: CreateUpdateAccountUseCase, getAccountsUseCase: GetAccountsUseCase) {
+        self.createUpdateAccountUseCase = createUpdateAccountUseCase
+        self.getAccountsUseCase = getAccountsUseCase
+    }
+    
+    func execute(category: DomainAccountModel, completion: @escaping (Result<[DomainAccountModel]>) -> Void) {
+        DispatchQueue.global().async {
+            self.createUpdateAccountUseCase.execute(request: category) { [weak self] _ in
+                self?.semaphore.signal()
```

Comment:
why we cannot call getAccoutUseCase here to get rid of semaphore?

Reply:
Yes, we can. I just wanted to get rid of callback indentations. But actually one more is not so bad. 
It will be much more scaleable if we use `Operation` and its dependencies though.